### DISCUSSION TITLE
Merge out side of the box

### DISCUSSION THREAD

**Discussion by ramirog89** [structurizr/cli#162] [General]
Hello!

I'm actually working in a big project where we've multiple teams and the intention is to use structurizr `onpremises` / `lite` and define a workflow to allow all team members to collaborate with the diagrams.

An actual concern we have, is about understanding the right flow to work with workspaces with manual layout in a collaborative way, considering that, a manual layout change generates a new `workspace.json` version with all the metadata, and resolving git conflicts on the `workspace.json` is very hard and not acceptable.

Hence I started looking ways to understand what we can do to solve next scenario:

Let's say that person 1 is on his local repository and:
- Updates the DSL
- Opens the structurizr/lite on the browser and updates the Layout manually
- Saves the changes (dsl and json files), commit and push.

Now person 2, has been working last days on the diagram and is 3 commits above remote HEAD and now:
- Person 2 want's to push, but git says he's 3 commits behind and need to run a `pull`.
- Person 2 pulls the changes, and has conflicts to resolve, `dsl` conflicts are easy..
- Now person 2 has to resolve `.json` file conflicts and this is not acceptable given both has too many changes and is impossible to merge them manually.
- Now the issue: if person 2 on his 3 commits has progressed and done several updates to the layout, he will has to somehow fix the conflicts (which is not acceptable) or otherwise, lose his work and delete his local `workspace.json` and re apply the manual layout updates he made before the pull which is tedious and generates frustration to the collaborators given that they will have to do the work twice.
- Once updates done, person 2 is able to commit and and push his changes, otherwise he would override remote changes with his local version.

First question: is this flow correct? is there a right way to manage this use case?

Second question: is it's possible to have the `merge` command in the `structurizr/cli` to run a `merge` with the remote and local changes... that way will be easier to fix the `workspace.json` merge.... considering that this will happen to several members on the team?...

I'm even considering to setup a random workspace to run the merge with my local and then pull the updated workspace, if works it could be like a workaround .. an ugly one, but yea.

Thank you very much for making this project!

Best.

**Comment by ramirog89** [comment]
Ok I found that the cli has a way to do the Merge.. not sure why I couldn't see it before, sorry for the message.

Either way, I found that is hard to work in a collaborative way with the manual layout given you'll always loose your local layout work when pulling changes from the Remote...

Looks like the only way to work with it is by using the `onpremise` which version the wroskpace everytime there's a push ... and locally, if I pull changes with new components.. I'll have to compare my local with the remote last version to maintain the layout. Right?

**Comment by simonbrowndotje** [comment]
This change should result in JSON that's easier to merge -> https://github.com/structurizr/java/issues/257
